deploy/00_deploy_protocol.js

```diff
@@ -0,0 +1,40 @@
+// deploy/00_deploy_my_contract.js
```

Comment:
I apologize as I'm just catching up on all this - I'm assuming this file is just middleware to deploy two different contracts but don't see the initial file - when you deploy does it also await for .deployed() as well?
In terms of writing contract to json would be something like this
```
const address = {};
address.metricToken = metricToken.address
address.chef = chef.address

  fs.writeFile(
    `${process.env.network}-address.json`,
    JSON.stringify(address),
     (err) => {
      if (err) throw err;
      console.log("address file created");
    }
  );
  ```

Reply:
hmm, so this file can be run using a hardhat plugin that introduces a new hardhat command `deploy`

Try running this `npx hardhat deploy --network ropsten` and you'll see a folder (which i added to gitignore) with all the deployment info! `deployments/ropsten`

Reply:
https://github.com/wighawag/hardhat-deploy :)

Reply:
ah that makes sense - will look into wasn't aware of it